### PR TITLE
Set a GroupName when creating new WinUI RadioButtons

### LINKED ISSUE #11418: Radio button loses initial 'IsChecked=True' value when another radio button also has 'IsChecked=True'
### Description

There is an issue when trying to use multiple radio buttons on a page.

If there are two radio buttons set to 'IsChecked=True' then only the last button in the view will remain true - despite them being given different GroupNames and in separate parent containers.
image

Initial thoughts are that it *appears* that all the radio buttons on the view are treated as one group (tried to give the control different GroupNames as shown in example) and therefore, as the view is rendered in order, only the last radio button with IsChecked=True will keep its initial value, earlier radio buttons will be set to be unchecked.
image

In Android, the expected result is shown. I have been unable to test the remaining platforms.
image

### Steps to Reproduce

Create a new solution.
Add to view multiple radio buttons with IsChecked=True

```
<VerticalStackLayout Spacing="25">

            <VerticalStackLayout>
                <RadioButton GroupName="First" Content="First" IsChecked="True"/>
            </VerticalStackLayout>

            <VerticalStackLayout>
                <RadioButton GroupName="Middle" Content="Middle" IsChecked="True"/>
            </VerticalStackLayout>

            <VerticalStackLayout>
                <RadioButton GroupName="Last" Content="Last" IsChecked="True"/>
            </VerticalStackLayout>

</VerticalStackLayout>
```

### Link to public reproduction project repository

https://github.com/andycambo/MauiRadioButtonBugExample

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 10.0.19041.0

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jstedfast**
### Description of Change

This change sets a unique GroupName on every WinUI RadioButton created so that different groups of MAUI RadioButtons are not assumed by WinUI to share the same group and thus prevent IsChecked values from being set properly.

### Issues Fixed

Fixes #11418

### CODE REVIEW

**Comment by hartez** [comment]
Will the random GroupNames cause problems later on with the implied grouping by container?

**Comment by jstedfast** [comment]
It doesn't seem to cause any problems from what I can tell.

**Comment by rmarinho** [comment]
@jstedfast do you think you can add a device test?

**Comment by jstedfast** [comment]
What would be the appropriate DeviceTests project to add tests for this to? Core.DeviceTests? Controls.DeviceTests?

**Comment by rmarinho** [comment]
I would say Core.DeviceTests

File: src/Controls/tests/DeviceTests/Elements/RadioButton/RadioButtonTests.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+			bool xplatIsChecked = isChecked;
+			var radioButton = new RadioButton() { IsChecked = xplatIsChecked };
+			bool expectedValue = isChecked;
+			var layoutFirst = new VerticalStackLayout();
+			var rdFirst = new RadioButton { GroupName = "FirstGroup", IsChecked = xplatIsChecked };
+			layoutFirst.Add(rdFirst);
+			layoutFirst.Add(new RadioButton { GroupName = "FirstGroup" });
+			layoutFirst.Add(new RadioButton { GroupName = "FirstGroup" });
+			var layoutSecond = new VerticalStackLayout();
+			var rdSecond = new RadioButton { GroupName = "SecondGroup", IsChecked = xplatIsChecked };
```
[the remark this answers is not in the document]
Comment (jstedfast) [reply src/Controls/tests/DeviceTests/Elements/RadioButton/RadioButtonTests.cs:26]: Yea, that's a good idea.

**Review by rmarinho — APPROVED** [review]
Awesome start Jeff thanks

**Pull request merged**